src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -457,7 +458,18 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
-func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
+func prepareAdditionalArguments(
+	developerMode bool, limits, requests corev1.ResourceList,
+) []string {
+	var smp int64 = 1
+	if !limits.Cpu().IsZero() {
+		limits.Cpu().RoundUp(0)
```

Comment:
I wonder if we should reduce the core count if each core would end up with <2GiB?

E.g., 2 cores requested. memory resource limit of 3GiB should give --smp=1. Maybe it's harder to reason about that a error in the log.

Reply:
Yeah, I took care of this